lib/cosmos/streams/tcpip_socket_stream.rb

```diff
@@ -48,9 +50,9 @@ def read
       # No read mutex is needed because there is only one stream procesor
       # reading
       begin
-        data = @read_socket.recv_nonblock(65535)
-        @raw_logger_pair.read_logger.write(data) if @raw_logger_pair
-      rescue Errno::EAGAIN, Errno::EWOULDBLOCK
+        @read_socket.read_nonblock(65535, @data)
```

Comment:
This prevents a new object from being created and garbage collected each time this is called. A huge win.